sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1090,6 +1090,18 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  val CACHE_DISABLE_CONFIGS =
+    buildConf("spark.sql.cache.disableConfigs")
+      .doc("Configurations needs to be turned off, to avoid regression for cached query, so that " +
+        "the outputPartitioning of the underlying cached query plan can be leveraged later.")
+      .version("3.2.0")
+      .stringConf
+      .toSequence
+      .checkValue(_.forall(v => sqlConfEntries.containsKey(v) &&
+        sqlConfEntries.get(v).defaultValue.exists(_.isInstanceOf[Boolean])),
+        "config should be boolean type")
+      .createWithDefault(Seq(ADAPTIVE_EXECUTION_ENABLED.key, AUTO_BUCKETED_SCAN_ENABLED.key))
```

Comment:
Any usecase to turn off these rules separately? I think it's okay just to use a boolean flag for this though.

Reply:
+1, I agree with starting with simpler config format if possible.